install-nimbus-cli.sh

```diff
@@ -0,0 +1,284 @@
+    curl -SsL "$DOWNLOAD_URL" -o "$NIMBUS_TMP_FILE"
+  elif [ "${HAS_WGET}" == "true" ]; then
+    wget -q -O "$NIMBUS_TMP_FILE" "$DOWNLOAD_URL"
+  fi
+}
+
+cleanup() {
+  if [[ -d "${NIMBUS_TMP_ROOT:-}" ]]; then
+    echoProgress "Cleaning up installation directory"
+    rm -rf "$NIMBUS_TMP_ROOT"
+  fi
+}
+
+success() {
+  echoProgress "Success!"
+}
+
+# runs the given command as root (detects if we are root already)
+runAsRoot() {
+  if [ $EUID -ne 0 ] && [ "$USE_SUDO" = "true" ]; then
+    sudo "${@}"
+  else
+    "${@}"
+  fi
+}
+
+# installFile installs the nimbus-cli binary.
+installFile() {
+  NIMBUS_TMP="$NIMBUS_TMP_ROOT/unzipped"
+  mkdir -p "$NIMBUS_TMP"
+  unzip -qj "$NIMBUS_TMP_FILE" -d "$NIMBUS_TMP"
+  local binary
+  if [[ "$OS" == "windows" ]] ; then
+    binary="${BINARY_NAME}.exe"
+  else
+    binary=$BINARY_NAME
+  fi
+  echoProgress "Preparing $binary for install"
+  chmod +x "$NIMBUS_TMP/$binary"
+  if [[ "$OS" == "darwin" ]] ; then
+    # Remove the quarantine bit from the binary
+    # to avoid the warning dialog
+    # https://eclecticlight.co/2017/12/11/xattr-com-apple-quarantine-the-quarantine-flag/
+    xattr -d com.apple.quarantine "$NIMBUS_TMP/$binary" 2>/dev/null && rc=$? || rc=$?
+    if [[ "$rc" == "0" ]] ; then
+      echoProgress Removed quarantine flag
+    else
+      echoInfo "Using for the first time may trigger a malicious software warning. Fix with: https://support.apple.com/en-us/guide/mac-help/mchleab3a043/mac"
+    fi
+  fi
+  runAsRoot cp "$NIMBUS_TMP/$binary" "$NIMBUS_INSTALL_DIR/$binary"
+  echoProgress "$binary installed into $NIMBUS_INSTALL_DIR"
+}
+
+# help provides possible cli installation arguments
+help () {
+  echo "install-nimbus-cli.sh"
+  echo
+  echo "Usage:"
+  echo -e "\t./install-nimbus-cli.sh [OPTIONS]"
+  echo
+  echo "Accepted cli options are:"
+  echo -e "\t--no-sudo                install without sudo"
+  echo -e "\t--directory DIRECTORY    install into the given directory"
+  echo -e "\t--host HOST              get from the given taskcluster host"
+  echo
+  echo -e "\t--debug                  be verbose in output"
+  echo -e "\t--help|-h                prints this help"
+}
+
+# fail_trap is executed if an error occurs.
+fail_trap() {
+  local result=$1
+  local lineNum=$2
+  cleanup
+  if [ "$result" != "0" ]; then
+    if [[ -n "$INPUT_ARGUMENTS" ]]; then
+      echoError "Failed to install $BINARY_NAME with the arguments provided: $INPUT_ARGUMENTS, at line number $lineNum"
+    else
+      echoError "Failed to install $BINARY_NAME, at line number $lineNum"
+    fi
+    echo -e "\tFor support, go to https://experimenter.info/nimbus-cli/install"
+  fi
+  exit "$result"
+}
+
+#Stop execution on any error
+trap 'fail_trap $? $LINENO' EXIT
+set -e
+
+# Set debug if desired
+if [ "${DEBUG}" == "true" ]; then
+  set -x
+fi
+
+# Parsing input arguments (if any)
+export INPUT_ARGUMENTS="${*}"
+set -u
+
+while (( "$#" )); do
+  case "$1" in
+    '--no-sudo')
+       USE_SUDO="false"
```

Comment:
Hmm does this get overwritten once `initInstallDirectory` runs? i.e even if the user sets `--no-sudo` we might run `sudo` anyways if the directory is protected (I also could be misfollowing the logic as well so feel free to correct me!)

Reply:
Yes, you're right.

I'm going to remove the `--no-sudo` option; I don't think it'll get any usage: it's a hangover from the get_helm script.